packages/api/src/functions/secureHandler.ts

```diff
@@ -0,0 +1,243 @@
+    super(message)
+  }
+}
+
+/**
+ * Class representing a WebhookVerificationError
+ * @extends WebhookError
+ */
+export class WebhookVerificationError extends WebhookError {
+  /**
+   * Create a WebhookVerificationError.
+   * @param {string} message - The error message
+   * */
+  constructor(message: string) {
+    super(message)
+  }
+}
+
+/**
+ * Generates a hash-based message authentication code from a secret.
+ *
+ * @param {string} secret - Secret key used to sign
+ * @return {string} - A HMAC
+ *
+ * @example
+ *
+ *    getHmac({ secret: 'MY_SECRET' })
+ */
+const getHmac = ({ secret }: { secret: string }) => {
+  if (typeof secret === 'undefined' || secret === '') {
+    throw new WebhookVerificationError(ERROR_MESSAGE)
+  }
+
+  return createHmac('sha256', secret)
+}
+
+/**
+ * Extracts signature from Lambda Event.
+ *
+ * @param {APIGatewayProxyEvent} event - The event that incudes the request details, like headers
+ * @return {string} - The header found in WEBHOOK_SIGNTAURE_HEADER
+ *
+ * @example
+ *
+ *    signatureFromEvent({ event: event })
+ */
+const signatureFromEvent = ({
+  event,
+}: {
+  event: APIGatewayProxyEvent
+}): string => {
+  return event?.headers?.[WEBHOOK_SIGNTAURE_HEADER.toLocaleLowerCase()]
+}
+
+/*
+ * The Redwood-Webhook-Signature header included in each signed event contains a timestamp and one or more signatures.
+ * The timestamp is prefixed by t=, and each signature is prefixed by a scheme. Schemes start with v, followed by an integer.
+ * Currently, the only valid live signature scheme is v1.
+ *
+ * The signed_payload string is created by concatenating:
+ *
+ * The timestamp (as a string)
+ * The character .
+ * The actual JSON payload (i.e., the request body)
+ *
+ * @param {string} body - The body payload to sign.
+ * @param {string} secret - The secret key used to sign. Defaults to WEBHOOK_SECRET.
+ * @param {number} timestamp - Timestamp in msec used to sign. Defaults to now.
+ * @return {string} - The signature
+ *
+ * @example
+ *
+ *    sign({ body: 'This is some content to sign.' })
+ */
+export const sign = ({
+  body,
+  secret = WEBHOOK_SECRET,
+  timestamp = Date.now(),
+}: {
+  body: string
+  secret: string
+  timestamp: number
+}): string => {
+  const hmac = getHmac({ secret })
+  hmac.update(timestamp + '.' + body)
+
+  return `t=${timestamp},v1=${hmac.digest('hex')}`
+}
+
+/**
+ * Verifies event is signed with a valid webhook signature.
+ * See verifySignature() for implementation rules.
+ *
+ * @param {APIGatewayProxyEvent} event - The event that incudes the request details, like headers.
+ * @param {VerifyOptions} options - Options for verifying the timestamp leeway.
+ * @return {boolean | WebhookVerificationError} - Returns true if the signature is verified, or raises WebhookVerificationError.
+ *
+ * @example
+ *
+ *    verify({ event: event, options: {} })*
+ */
+export const verify = ({
+  event,
+  options,
+}: {
+  event: APIGatewayProxyEvent
+  options?: VerifyOptions
```

Comment:
Perhaps add an `method` or `algorithm` to the `VerifyOptions` and if:

* Redwood Signature then `verifySignature`
* Otherwise use `crypto.createVerify` and its `verify` with algorithm -- though will need to pass the secret, too